### ISSUE TITLE
Gem fails db:create and db:drop if you reference tagged model in environment

### ISSUE THREAD

**Issue by vonagam** [tmiyamon/acts-as-taggable-array-on#15]
Commit 90e427816839529d1e415a4245a4218ae63e3fe9 moved database type reflection from runtime (scope) to load time (class level). Now, if you reference model with `acts_as_taggable_array_on` in environment and you don't have database - it fails.

In my case i had devise in routes, which referenced User model, which had tags on it. And routes are loaded in environment for db tasks.

**Comment by skatkov** [comment]
Thanks for reporting!

**Comment by vonagam** [comment]
I have reproduced it manually with new empty rails app (before opening issue):
1) create new rails app
2) add "acts-as-taggable-array-on" gem
3) add model with `acts_as_taggable_array_on :tags` (let's call it `Dummy`)
4) reference this model in routes (just write `Dummy` inside routes)
5) try to run db:create task
6) it will fail because of `columns_hash[tag_name.to_s].type`

**Comment by skatkov** [comment]
I believe i managed to reproduce this error and fixed it. Can you verify that it works for you too @vonagam  ?
#20

**Comment by vonagam** [comment]
This fix addresses case with missing column (when you added code to class before running migration).

This fix does not address case with missing table (as in first db:create).

`Temping.create :dummy` creates table, when you run db:create first time there will be no table.

**Comment by skatkov** [comment]
@vonagam can you attack an error your seeing, so I could be sure that I correctly reproduced your issue?

**Comment by vonagam** [comment]
Just commend out `Temping.create :dummy`. And it will be as i wrote before:
```
context "without database table" do
  it "fails on class method call" do
    class Dummy < ActiveRecord::Base; end
    Dummy.acts_as_taggable_array_on :tags
  end
end
```

**Comment by skatkov** [comment]
Should be fixed in 0.5.1 version.
